internal/provider/datasources/block_test.go

```diff
@@ -0,0 +1,73 @@
+package datasources_test
+
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/prefecthq/terraform-provider-prefect/internal/testutils"
+)
+
+func fixtureAccBlockByName(name string) string {
+	return fmt.Sprintf(`
+data "prefect_workspace" "evergreen" {
+	handle = "github-ci-tests"
+}
+
+resource "prefect_block" "%s" {
+  name      = "%s"
+  type_slug = "secret"
+
+  data = jsonencode({
+    "someKey" : "someValue"
+  })
+
+  account_id = "%s"
+  workspace_id = data.prefect_workspace.evergreen.id
+}
+
+data "prefect_block" "my_existing_secret_by_id" {
+  id = prefect_block.%s.id
```

Comment:
you might need the `workspace_id` here too

Reply:
TIL, thanks - that got me further. Now seeing:

```
    block_test.go:57: Step 1/2 error: Error running pre-apply plan: exit status 1
        
        Error: Error refreshing block state
        
          with data.prefect_block.my_existing_secret_by_name,
          on terraform_plugin_test.tf line 35, in data "prefect_block" "my_existing_secret_by_name":
          35: data "prefect_block" "my_existing_secret_by_name" {
        
        Could not read block, unexpected error: status code 404 Not Found,
        error={"detail":"Block document not found"}
```

That portion saying `running pre-apply plan` makes me think the test didn't get a chance to actually apply the plan yet, so expected that the resource won't be there.

Reply:
Fixed by specifying `depends_on` in the datasources.